subsys/bluetooth/host/hci_core.c

```diff
@@ -834,6 +873,15 @@ static void le_remote_feat_complete(struct net_buf *buf)
 		       sizeof(conn->le.features));
 	}
 
+	if (BT_FEAT_LE_DLE(bt_dev.le.features)) {
```

Comment:
As a peripheral if peer does not support slave initiated feature request, we still want to check DLE is supported.